pkgs/development/rocm-modules/6/clr/default.nix

```diff
@@ -135,88 +158,140 @@ stdenv.mkDerivation (finalAttrs: {
       --replace "install(PROGRAMS \''${HIPCC_BIN_DIR}/hipconfig.bat DESTINATION bin)" ""
 
     substituteInPlace hipamd/src/hip_embed_pch.sh \
-      --replace "\''$LLVM_DIR/bin/clang" "${clang}/bin/clang"
-
-    # https://lists.debian.org/debian-ai/2024/02/msg00178.html
-    substituteInPlace rocclr/utils/flags.hpp \
-      --replace-fail "HIP_USE_RUNTIME_UNBUNDLER, false" "HIP_USE_RUNTIME_UNBUNDLER, true"
+      --replace "\''$LLVM_DIR/bin/clang" "${hipClangPath}/bin/clang"
 
     substituteInPlace opencl/khronos/icd/loader/icd_platform.h \
       --replace-fail '#define ICD_VENDOR_PATH "/etc/OpenCL/vendors/";' \
                      '#define ICD_VENDOR_PATH "/run/opengl-driver/etc/OpenCL/vendors/";'
+
+    # new unbundler has better error messages, defaulting it on
+    substituteInPlace rocclr/utils/flags.hpp \
+      --replace-fail "HIP_ALWAYS_USE_NEW_COMGR_UNBUNDLING_ACTION, false" "HIP_ALWAYS_USE_NEW_COMGR_UNBUNDLING_ACTION, true"
   '';
 
   postInstall = ''
+    chmod +x $out/bin/*
     patchShebangs $out/bin
 
-    # hipcc.bin and hipconfig.bin is mysteriously never installed
-    cp -a ${hipcc}/bin/{hipcc.bin,hipconfig.bin} $out/bin
+    cp ${amdclang}/bin/* $out/bin/
+    cp ${amdclangxx}/bin/* $out/bin/
 
-    wrapProgram $out/bin/hipcc.bin ${lib.concatStringsSep " " wrapperArgs}
-    wrapProgram $out/bin/hipconfig.bin ${lib.concatStringsSep " " wrapperArgs}
+    wrapProgram $out/bin/hipcc ${lib.concatStringsSep " " wrapperArgs}
+    wrapProgram $out/bin/hipconfig ${lib.concatStringsSep " " wrapperArgs}
     wrapProgram $out/bin/hipcc.pl ${lib.concatStringsSep " " wrapperArgs}
     wrapProgram $out/bin/hipconfig.pl ${lib.concatStringsSep " " wrapperArgs}
 
+    mkdir -p $out/nix-support/
+    echo '
+    CORE_LIM="''${NIX_BUILD_CORES:-1}"
+    if ((CORE_LIM <= 0)); then
+        guess=$(nproc 2>/dev/null || true)
+        ((CORE_LIM = guess <= 1 ? 1 : guess))
+        ((CORE_LIM = CORE_LIM >= 3 ? 3 : CORE_LIM))
+    fi
+    CORE_LIM=$(( ''${NIX_LOAD_LIMIT:-''${CORE_LIM:-$(nproc)}} / 2 ))
+    # Set HIPCC_JOBS with min and max constraints
+    export HIPCC_JOBS=$CORE_LIM
+    export HIPCC_JOBS_LINK=$CORE_LIM
+    export CFLAGS="''${CFLAGS:-} -parallel-jobs=$CORE_LIM"
+    export CXXFLAGS="''${CXXFLAGS:-} -parallel-jobs=$CORE_LIM"
```

Comment:
I assume `-parallel-jobs=$CORE_LIM` is just for testing purposes for now?

Reply:
Defaulted it on because some packages had very slow single-core bottlenecks without it due to spending a long time linking one binary or building one large templatized kernel for multiple arches in one amdclang invocation.

Probably best to drop setting this in CLR and turn it on for exactly those packages.

Reply:
Another alternative is setting `-DHIP_CLANG_NUM_PARALLEL_JOBS=XX` should work for most AMD repos and won't work for most third-party projects that that use amdclang iirc.

Reply:
Removed from clr